# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/HTMLEditor.h
# Commit: 42ed78fa08b2
# Full Hash: 42ed78fa08b226f3de66ce279740432bcd393a84
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2021-08-27 09:43:40
# Regressor Bug: 1727185
# File Overlap Count: 1
# Description:
#   Bug 1727185 - part 2: Add removable node checks for more callers of `DeleteNodeTransaction::MaybeCreate()` r=m_kato
#   
#   I just missed some callers of it.  And I also make
#   `EditorBase::DeleteNodeTransaction()` a virtual method and `HTMLEditor` override
#   it.  Then, even if it's called by helper classes, the `HTMLEditor` version is
# ==============================================================================

diff -r f79cc68cd2fe -r 42ed78fa08b2 editor/libeditor/HTMLEditor.h
--- a/editor/libeditor/HTMLEditor.h	Fri Aug 27 00:26:30 2021 +0000
+++ b/editor/libeditor/HTMLEditor.h	Fri Aug 27 00:49:35 2021 +0000
@@ -755,12 +755,12 @@
 
   /**
    * DeleteNodeWithTransaction() removes aContent from the DOM tree if it's
-   * modifiable.  Note that this is not an override of same method of
-   * EditorBase.
+   * modifiable.
    *
    * @param aContent    The node to be removed from the DOM tree.
    */
-  MOZ_CAN_RUN_SCRIPT nsresult DeleteNodeWithTransaction(nsIContent& aContent);
+  MOZ_CAN_RUN_SCRIPT nsresult
+  DeleteNodeWithTransaction(nsIContent& aContent) final;
 
   /**
    * DeleteTextWithTransaction() removes text in the range from aTextNode if
